internal/client/opendota/match.go

```diff
@@ -8,31 +8,40 @@ import (
 	"net/http"
 )
 
+const (
+	DefaultBaseURL = "https://api.opendota.com/api"
+)
+
 var (
 	ErrFetchMatchDetail = fmt.Errorf("failed to fetch match detail")
 )
 
-type OpenDotaAPI struct {
+type API struct {
 	httpClient *http.Client
 	BaseURL    string
 }
 
-func NewMatchAPI(httpClient *http.Client, BaseURL string) *OpenDotaAPI {
-	return &OpenDotaAPI{
+func NewAPI(httpClient *http.Client, BaseURL string) *API {
+	return &API{
 		httpClient: httpClient,
 		BaseURL:    BaseURL,
 	}
 }
 
-func (m *OpenDotaAPI) FetchMatchDetail(ctx context.Context, matchID int64) (MatchDetail, error) {
-	matchDetailURL := fmt.Sprintf("%s/matches/%d", m.BaseURL, matchID)
+// NewDefaultAPI creates a OpenDotaAPI with default params.
+func NewDefaultAPI() *API {
+	return NewAPI(http.DefaultClient, DefaultBaseURL)
+}
+
+func (api *API) FetchMatchDetail(ctx context.Context, matchID int64) (MatchDetail, error) {
+	matchDetailURL := fmt.Sprintf("%s/matches/%d", api.BaseURL, matchID)
```

Comment:
I missed this, `url` short form instead of `matchDetailURL` and `id` instead of `matchID` is should be clear enough.